visualization/app/codeCharta/util/fileExtensionCalculator.ts

```diff
@@ -90,18 +79,19 @@ export class FileExtensionCalculator {
 		return {
 			fileExtension,
 			absoluteMetricValue: metricValue,
-			relativeMetricValue: null,
+			relativeMetricValue: 0,
 			color: null
 		}
 	}
 
-	private static getSumOfAllMetrics(distribution: MetricDistribution[]) {
-		return distribution.reduce((partialSum, a) => partialSum + a.absoluteMetricValue, 0)
-	}
-
 	static estimateFileExtension(fileName: string) {
-		if (fileName.includes(".")) {
-			return fileName.split(".").reverse()[0].toLowerCase()
+		const lastDotPosition = fileName.lastIndexOf(".")
+		if (lastDotPosition > 0) {
```

Comment:
This is a bug fix: it does not match entries without extension anymore (files that start with a dot and that do not have any other character are extension-less).